core/diagnostics/DiagnosticScenarios/Controllers/DiagnosticScenarios.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading;
+
+namespace testwebapi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class DiagScenarioController : ControllerBase
+    {
+        object o1 = new object();
+        object o2 = new object();
+
+        private static Processor p = new Processor();
+
+        [HttpGet]
+        [Route("deadlock/")]
+        public ActionResult<string> deadlock()
+        {
+            (new System.Threading.Thread(() => {
+                DeadlockFunc();
+            })).Start();
+
+            Thread.Sleep(5000);
+
+            Thread[] threads = new Thread[300];
+            for(int i=0; i<300;i++)
+            {
+                (threads[i] = new Thread(() => {
+                    lock (o1) {Thread.Sleep(100);}
+                })).Start();
+            }
+
+            foreach(Thread thread in threads)
+            {
+                thread.Join();
+            }
+
+            return "success:deadlock";
+        }
+
+        private void DeadlockFunc()
+        {
+            lock (o1)
+            {
+                (new Thread(() => {
+                    lock (o2) { Monitor.Enter(o1); }
+                })).Start();
+
+                Thread.Sleep(2000);
+                Monitor.Enter(o2);
+            }
+        }
+
+        [HttpGet]
+        [Route("memspike/{seconds}")]
+        public ActionResult<string> memspike(int seconds)
+        {
+            Stopwatch watch=new Stopwatch();
+            watch.Start();
+
+            while(true)
+            {
+                p = new Processor();
+                watch.Stop();
+                if(watch.ElapsedMilliseconds > seconds*1000)
+                    break;
+                watch.Start();
+
+                int it = (200000*1000) / 100;
+                for(int i=0; i<it; i++)
+                {
+                    p.ProcessTransaction(new Customer(Guid.NewGuid().ToString()));
+                }
+
+                Thread.Sleep(5000);	// Sleep for 5 seconds before cleaning up
+
+                // Cleanup
+                p = null;
+                GC.Collect();
+                GC.Collect();
```

Comment:
Not sure if lines 84 and 85 are duplicated by mistake or there is a reason for that.

Reply:
I wouldn't be surprised if it wasn't required.

Reply:
I can't imagine why you'd want to call `GC.Collect` multiple times in succession. @Maoni0, are there any scenarios in which GC.Collect would be called successively?

Reply:
`GC.Collect()` is not required to do an exhaustive collection.

I know in fixing a customer's `AssemblyLoadContext` unloading test, we had to put the `GC.Collect()` in a loop to waiting for the `AssemblyLoadContext` unload.

Reply:
that's because AssemblyLoadContext has a finalizer. you should do 

GC.Collect();
GC.WaitForPendingFinalizers();
GC.Collect();